app/domain/authentication/authn_iam/authenticator.rb

```diff
@@ -54,7 +54,27 @@ def extract_relevant_data(response)
 
       # Call to AWS STS endpoint using the provided authentication header
       def attempt_signed_request(signed_headers)
-        aws_request = URI("https://#{signed_headers['host']}/?Action=GetCallerIdentity&Version=2011-06-15")
+        sts_host = signed_headers['host'] || extract_sts_host(signed_headers)
```

Comment:
I'd recommend we move the signed header resolution into the `extract_sts_host` method. This allows us to keep all logic related to 

```ruby
sts_host = signed_headers['host'] || extract_sts_host(signed_headers)
```
Then we push the logic for checking header value or attempting to extract the region from the credential:
```ruby
# Extract AWS region from the authorization header's credential string, i.e.:
# Credential=AKIAIOSFODNN7EXAMPLE/20220830/us-east-1/sts/aws4_request
def extract_sts_host(signed_headers)
  return signed_headers['host'] if signed_headers['host'].present?

  region = signed_headers['authorization'].match(%r{Credential=[^/]+/[^/]+/([^/]+)/})&.captures&.first
  raise(Errors::Authentication::AuthnIam::InvalidAWSHeaders, 'Failed to extract AWS region from authorization header') unless region

  "sts.#{region}.amazonaws.com"
end
```